enhancements/etcd/disaster-recovery-with-ceo.md

```diff
@@ -19,86 +19,200 @@ see-also:
+simplify other existing scripts for backing up and restoring the cluster state. 
 
-## User Stories [optional]
+With the simplification of the scripts, different disaster recovery scenarios are documented properly to utilize the
+simplified scripts along with other Openshift utility commands to achieve the recovery of the cluster. 
 
 ## Implementation Plan
-
-1. Make changes to the scripts as needed
-2. Modify CEO to provide the information needed for the DR scripts to work.
+###  Transfer the ownership of the DR files from the MCO to CEO
+All etcd-scripts should be removed from MCO, and be generated along with the static-pod-resources as configmaps. The 
+scripts should be copied to /usr/local/bin in the init container for a convenient access to the admin user.
+    
+###  Remove the deprecated DR scripts
+The scripts for adding a new etcd member; removing an etcd member; and recovering an etcd-member are no longer needed,
+and should be deleted from the repository.
+    
+### Backing up the cluster state
+The script for backing up the cluster state should back up static-pod-resources for etcd, kube-apiserver, 
+kubecontrollermanager and kube-scheduler, along with a copy of the etcd snapshot database. The reason for backing up
+the static pod resources is to match the static pods state on disk with what is in etcd,  when we restore. Furthermore,
+when etcd data is encrypted, the encryption keys are stored in static pod resources, and restoring the etcd data without
+restoring the corresponding key will make the etcd data unusable.
+    
+### Restoring the cluster state from backup 
+This solution handles situations where you want to restore your cluster to a previous state, for example, if something
+critical got deleted accidentally. This process is also used for recovering from a loss of quorum.
+    
+The script for restoring to a previous cluster state using a backup file will restore the etcd database from the backup
+along with the static pod resources for the entire control plane (etcd, kube-apiserver, kubecontrollermanager and
+kube-scheduler). See the implementation details for more information.
+    
+### Removal of a member from the etcd cluster
+In the past, we used a script to remove a member from the etcd cluster. In this release, the remove operation will be
+documented to invoke the `etcdctl member remove` command directly on the newly created `etcdctl` container.
+    
+### Recovering a single unhealthy member
+See the implementation details below.
+    
+### Recovering from a loss of quorum
+Recovering from a loss of quorum requires restoring from a previous backup. See the implementation details for
+`Restoring the cluster from backup`.
+   
+### Recovery of a member with a bad data-dir.
+If you have a majority of your masters still available and have an etcd quorum, then the following procedure should
+restore the data-dir:
+1. Stop etcd static pod by moving the etcd pod yaml out of kubelet manifest directory.
```

Comment:
this shouldn't be necessary.  won't it be crashlooping?  if so, you can simply delete the data dir.